sklearn/metrics/pairwise.py

```diff
@@ -74,9 +74,10 @@ def check_pairwise_arrays(
     Y,
     *,
     precomputed=False,
-    dtype=None,
+    dtype="infer_float",
```

Comment:
I changed the default ``dtype`` in ``check_pairwaise_arrays`` and also the behavior of ``dtype=None`` because it was inconsistent with the behavior of ``dtype=None`` in ``check_array`` (i.e. preserve dtype).

Now:
``dtype="infer_float"`` has the same behavior as previously ``dtype=None``, that is convert to appropriate float.
``dtype=None`` preserves the input dtype.

``check_pairwaise_arrays`` is private and I checked that scikit-learn doesn't use the dtype parameter anywhere else than in ``pairwise_distances``. That's why I directly changed the default and the behavior. Let me know if you're okay with that.